### PR TITLE
Upgrade CI to ROCm5.7

### PR DESCRIPTION
**Pull request by rraminen**
This PR is to upgrade CI to ROCm5.7

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @zstreet87

### CODE REVIEW

**Comment by rraminen** [comment]
Failures from https://github.com/pytorch/pytorch/actions/runs/6416057584/job/17420731822?pr=110465
```
test_quantization.py::TestQuantizePT2EQAT::test_qat_conv_bn_relu_fusion_cuda <- test/quantization/pt2e/test_quantize_pt2e_qat.py FAILED [0.9166s] [ 69%]
nn/test_convolution.py::TestConvolutionNN::test_Conv2d_groups_nobias FAILED [0.0173s] [  0%]
nn/test_convolution.py::TestConvolutionNN::test_Conv2d_groups_nobias_v2 FAILED [0.0164s] [  1%]
nn/test_convolution.py::TestConvolutionNNDeviceTypeCUDA::test_Conv2d_naive_groups_cuda_float16 FAILED [0.0079s] [ 10%]

```

Failures from https://github.com/pytorch/pytorch/actions/runs/6416057584/job/17420732445?pr=110465
```
dynamo/test_logging.py::LoggingTests::test_graph_breaks <- ../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/testing/_internal/logging_utils.py FAILED [0.0570s] [ 43%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_ceil_fastpath_inplace_cuda_bfloat16 FAILED [0.0184s] [ 45%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_ceil_fastpath_inplace_cuda_float64 FAILED [0.0182s] [ 45%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_ceil_fastpath_outplace_cuda_float32 FAILED [0.0227s] [ 45%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_ceil_fastpath_outplace_cuda_float64 FAILED [0.0224s] [ 45%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_max_fastpath_inplace_cuda_bfloat16 FAILED [0.1214s] [ 46%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_max_fastpath_inplace_cuda_int8 FAILED [0.1101s] [ 47%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_max_fastpath_outplace_cuda_bfloat16 FAILED [0.1514s] [ 47%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_max_fastpath_outplace_cuda_float16 FAILED [0.1532s] [ 47%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_max_fastpath_outplace_cuda_float32 FAILED [0.2408s] [ 47%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_max_fastpath_outplace_cuda_float64 FAILED [0.1308s] [ 47%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_max_fastpath_outplace_cuda_int16 FAILED [0.1919s] [ 47%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_clamp_min_fastpath_outplace_cuda_uint8 FAILED [0.1812s] [ 49%] 
test_foreach.py::TestForeachCUDA::test_parity__foreach_cos_fastpath_inplace_cuda_float16 FAILED [0.0178s] [ 51%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_cos_fastpath_inplace_cuda_float32 FAILED [0.0185s] [ 51%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_cos_fastpath_inplace_cuda_float64 FAILED [0.0179s] [ 51%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_cos_fastpath_outplace_cuda_float64 FAILED [0.0225s] [ 51%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_cosh_fastpath_outplace_cuda_complex64 FAILED [0.0259s] [ 52%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_div_fastpath_inplace_cuda_bfloat16 FAILED [0.1661s] [ 52%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_div_fastpath_inplace_cuda_complex128 FAILED [0.1334s] [ 52%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_div_fastpath_inplace_cuda_complex64 FAILED [0.1161s] [ 52%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_div_fastpath_inplace_cuda_float16 FAILED [0.1547s] [ 52%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_div_fastpath_inplace_cuda_float64 FAILED [0.1403s] [ 53%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_div_fastpath_outplace_cuda_complex128 FAILED [0.1383s] [ 53%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_exp_fastpath_inplace_cuda_complex64 FAILED [0.0212s] [ 56%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_floor_fastpath_inplace_cuda_float32 FAILED [0.0189s] [ 58%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_log10_fastpath_inplace_cuda_float32 FAILED [0.0208s] [ 63%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_maximum_fastpath_inplace_cuda_float32 FAILED [0.1432s] [ 67%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_maximum_fastpath_inplace_cuda_float64 FAILED [0.1442s] [ 67%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_maximum_fastpath_outplace_cuda_bfloat16  FAILED [0.1700s] [ 67%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_maximum_fastpath_outplace_cuda_float16 FAILED [0.1434s] [ 67%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_minimum_fastpath_inplace_cuda_bfloat16 FAILED [0.1241s] [ 68%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_minimum_fastpath_inplace_cuda_float64 FAILED [0.1507s] [ 68%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_minimum_fastpath_inplace_cuda_uint8 FAILED [0.1026s] [ 69%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_minimum_fastpath_outplace_cuda_bfloat16 FAILED [0.1354s] [ 69%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_minimum_fastpath_outplace_cuda_float16 FAILED [0.1343s] [ 69%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_minimum_fastpath_outplace_cuda_float32 FAILED [0.1749s] [ 69%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_minimum_fastpath_outplace_cuda_int16 FAILED [0.1479s] [ 69%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_inplace_cuda_bool FAILED [0.1622s] [ 70%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_inplace_cuda_complex128 FAILED [0.2973s] [ 70%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_inplace_cuda_float32 FAILED [0.1646s] [ 70%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_inplace_cuda_float64 FAILED [0.1433s] [ 70%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_inplace_cuda_int16 FAILED [0.4073s] [ 70%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_outplace_cuda_complex128 FAILED [0.1578s] [ 71%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_outplace_cuda_complex64 FAILED [0.1865s] [ 71%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_mul_fastpath_outplace_cuda_uint8 FAILED [0.1835s] [ 71%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_pow_fastpath_inplace_cuda_float16 FAILED [0.1435s] [ 75%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_pow_fastpath_outplace_cuda_float16 FAILED [0.1715s] [ 75%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_pow_fastpath_outplace_cuda_float32 FAILED [0.1868s] [ 75%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_sin_fastpath_outplace_cuda_complex64 FAILED [0.0257s] [ 80%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_sinh_fastpath_inplace_cuda_complex128 FAILED [0.0218s] [ 81%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_sub_fastpath_inplace_cuda_complex128 FAILED [0.1461s] [ 83%]
test_foreach.py::TestForeachCUDA::test_parity__foreach_tan_fastpath_inplace_cuda_complex128 FAILED [0.0214s] [ 85%]
```

**Comment by jithunnair-amd** [comment]
@rraminen The test_foreach*fastpath* failures are probably all related to the Kineto fix coming up in: https://github.com/pytorch/pytorch/pull/110849
The Conv2d_*groups failures are likely all related to the tolerance adjustment in: https://github.com/pytorch/pytorch/pull/110656

**Comment by jithunnair-amd** [comment]
Re: The Conv2d_*groups failures are likely all related to the tolerance adjustment in: https://github.com/pytorch/pytorch/pull/110656
Will add github issues to skip these to unblock ROCm5.7 CI upgrade PR.

**Comment by rraminen** [comment]
The tests, 
`dynamo/test_logging.py::LoggingTests::test_graph_breaks  test_quantization.py::TestQuantizePT2EQAT::test_qat_conv_bn_relu_fusion_cuda`
are not failing in the latest CI logs.

https://github.com/pytorch/pytorch/actions/runs/6564510115/job/17835021627?pr=110465
https://github.com/pytorch/pytorch/actions/runs/6564510115/job/17835021857?pr=110465

**Comment by rraminen** [comment]
Added https://github.com/pytorch/pytorch/pull/110465/commits/7e2451bff021c0a2ad4ce335be74f02aa34db1ae commit to skip the below unit tests on ROCm.

nn/test_convolution.py::TestConvolutionNN::test_Conv2d_groups_nobias FAILED [0.0173s] [  0%]
nn/test_convolution.py::TestConvolutionNN::test_Conv2d_groups_nobias_v2 FAILED [0.0164s] [  1%]
nn/test_convolution.py::TestConvolutionNNDeviceTypeCUDA::test_Conv2d_naive_groups_cuda_float16 FAILED [0.0079s] [ 10%]

**Comment by rraminen** [comment]
There are no pyt unit test failures on ROCm. 

cc: @pruthvistony

**Comment by pruthvistony** [comment]
> There are no pyt unit test failures on ROCm.
> 
> cc: @pruthvistony

@rraminen ,
ciflow/rocm wasn't added which is newly required to be added to trigger the ROCm SGPU UTs. Added it now, lets wait for the results.

**Comment by pruthvistony** [comment]
Failing UTs on ROCm are skipped. Re-triggering CI to get a green signal.

**Comment by rraminen** [comment]
https://github.com/pytorch/pytorch/actions/runs/6725987468/job/18282551650?pr=110465
83 tests from test_foreach.py::TestForeachCUDA suite fail


https://github.com/pytorch/pytorch/actions/runs/6725987468/job/18282551762?pr=110465  one failure 
`dynamo/test_trace_rules.py::TraceRuleTests::test_torch_name_rule_map FAILED [0.1186s]`

**Comment by pruthvistony** [comment]
> https://github.com/pytorch/pytorch/actions/runs/6725987468/job/18282551650?pr=110465 83 tests from test_foreach.py::TestForeachCUDA suite fail
> 
> https://github.com/pytorch/pytorch/actions/runs/6725987468/job/18282551762?pr=110465 one failure `dynamo/test_trace_rules.py::TraceRuleTests::test_torch_name_rule_map FAILED [0.1186s]`

@rraminen ,
All the test_foreach.py test_parity_** UT were passing on ROCm when added https://github.com/pytorch/pytorch/pull/107869 on ROCm 5.6. Need to find if ROCm 5.7 is causing the break or other upstream changes to torch/kineto.

**Comment by pruthvistony** [comment]
Regarding the failing ROCm distributed test - 

_FAILED [6.4297s] distributed/test_compute_comm_reordering.py::TestComputeCommReorderingMultiProc::test_reorder_compute_for_overlap - RuntimeError: Process 0 exited with error code 10 and exception:_

Previously this UT passed - 
https://hud.pytorch.org/pytorch/pytorch/pull/110465?sha=0d34562d9ab72af3c8fe75fc4e2ebdec5541d491
So considering this failure as flaky and regarding the remaining 9 test jobs failures are all unrelated to this change.

File: test/nn/test_convolution.py [hunk 1]
```diff
@@ -481,6 +481,7 @@ def test_ConvTranspose2d_half_cublas_gemm(self):
     # For https://github.com/pytorch/pytorch/pull/1273
     # Almost identical to the above `test_Conv2d_naive_groups`
     @torch.backends.cudnn.flags(enabled=True, benchmark=False)
+    @unittest.skipIf(TEST_WITH_ROCM, "Skipped on ROCm")
```
Comment (pruthvistony) [inline test/nn/test_convolution.py:484]: Better to update the skip reason saying "Skipped on ROCm, since it is failing on ROCm 5.7"

**Review by pruthvistony — APPROVED** [review]
LGTM

**Comment by pruthvistony** [comment]
@jeffdaily @malfet,
Please review this PR.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**